flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/allocator/SlotAllocator.java

```diff
@@ -54,14 +58,35 @@ public interface SlotAllocator {
     Optional<? extends VertexParallelism> determineParallelism(
             JobInformation jobInformation, Collection<? extends SlotInfo> slots);
 
+    /**
+     * Same as {@link #determineParallelism(JobInformation, Collection)} but additionally determine
+     * assignment of slots to execution slot sharing groups.
+     */
+    default Optional<JobSchedulingPlan> determineParallelismAndCalculateAssignment(
+            JobInformation jobInformation,
+            Collection<? extends SlotInfo> slots,
+            @Nullable ExecutionGraph previousExecutionGraph) {
+        return determineParallelismAndCalculateAssignment(
+                jobInformation,
+                slots,
+                JobAllocationsInformation.fromGraph(
+                        previousExecutionGraph,
+                        StateSizeEstimates.fromGraph(previousExecutionGraph)));
+    }
```

Comment:
Should we have this method? We can also simplify the JobAllocationInformation construction to `JobAllocationsInformation.fromGraph(previousExecutionGraph)`, which seems to eliminate the need even more.

I was looking at the mock implementation and realized that this would be hard to mock correctly.

Reply:
Right, mocking was the reason why I introduced the 2nd method. 
I agree, now with `JobAllocationsInformation` that method can be directly called from the scheduler. 
I'll remove this method with `ExecutionGraph`.

Reply:
Done in 106dbceed065c38347f33e7c9320e0b0b904e0ba.